entrypoint.sh

```diff
@@ -66,8 +66,24 @@ while http_status=$(curl -o /dev/null -w '%{http_code}' -sL "${NOTICE_URL}"); [
   sleep "${POLL_INTERVAL}"
 done
 
+# Similar Worker Count
+# For added confidence that terminated worker nodes are being replaced, we can add a count of similarly-tagged EC2 instances to our message
+# Enable and filter the query by setting SWC_TAG_NAME, e.g. "Name", and SWC_TAG_VALUES, e.g. "eks-worker-spot,eks-worker-ondemand"
+# Note: a spot instance remains in "running" state throughout the termination notice period, after which the shutdown signal is sent
+SWC_TAG_NAME=${SWC_TAG_NAME:-""}
+SWC_TAG_VALUES=${SWC_TAG_VALUES:-""}
+SWC_TIMEOUT=${SWC_TIMEOUT:-5}
+SWC_MESSAGE="Disabled"
+if [ "${SWC_TAG_NAME}" != "" ] && [ "${SWC_TAG_VALUES}" != "" ]; then
+  SWC_COUNT=$(aws ec2 --cli-read-timeout $SWC_TIMEOUT --output text --query "Reservations[*].Instances[*] | length(@)" --region $REGION describe-instances --filters "Name=instance-state-name,Values=running" "Name=tag:$SWC_TAG_NAME,Values=$SWC_TAG_VALUES" 2>/dev/null)
+  case $SWC_COUNT in
+    ''|*[!0-9]*) SWC_MESSAGE="Failed" ;;
+    *) SWC_MESSAGE="$((SWC_COUNT-1))" ;;
+  esac
+fi
+
 echo "$(date): ${http_status}"
-MESSAGE="Spot Termination${CLUSTER_INFO}: ${NODE_NAME}, Instance: ${INSTANCE_ID}, Instance Type: ${INSTANCE_TYPE}, AZ: ${AZ}"
+MESSAGE="Spot Termination${CLUSTER_INFO}: ${NODE_NAME}, Instance: ${INSTANCE_ID}, Instance Type: ${INSTANCE_TYPE}, AZ: ${AZ}, Surviving Worker Count: ${SWC_MESSAGE}"
```

Comment:
It seems like even if the variables aren't set, you'd get `Surviving Worker Count:` in the output.
If we don't set variables, we shouldn't get that output, makes it messy.
Am I reading it correctly?

Reply:
I definitely don't mind adding the feature, but we need to be clean in the use case when the feature is not being activated. :)

Reply:
Sorry for the delay, and thanks for the feedback. I plan on making the changes you suggest as soon as I find some time.